### PR TITLE
Issue888 scaling window

### LINKED ISSUE #888: Scaling of window
Zones of large buildings often have a series of identical windows with the same orientation. In case of, for example, box-shading, the windows can not simply be lumped into one.

I propose therefore to add a parameter to the window model which makes it possible to simulate _n_ windows with a single model.

I'll propose an implementation.

### PR DESCRIPTION
**Pull request by damienpicard**
This closes #888

### CODE REVIEW

**Comment by damienpicard** [comment]
@Mathadon can you add reference results for `IDEAS\IDEAS\Resources\Scripts\Dymola\Buildings\Components\Examples\ScalingWindow` please?

**Comment by Mathadon** [comment]
The `EnergyConservationValidation` fails. This has to be checked.

File: IDEAS/Buildings/Components/InternalWall.mo [hunk 1]
```diff
@@ -202,6 +204,12 @@ We assume that the value of <code>A</code> excludes the surface area of the cavi
 </html>", revisions="<html>
 <ul>
 <li>
+August 10, 2018 by Damien Picard:<br/>
+Set nWin final to 0 as this should only be used for windows.
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/InternalWall.mo]: should be '1'

File: IDEAS/Buildings/Components/OuterWall.mo [hunk 2]
```diff
@@ -242,6 +243,12 @@ The correct shading parameter values should then be passed through the redeclara
 </html>", revisions="<html>
 <ul>
 <li>
+August 10, 2018 by Damien Picard:<br/>
+Set nWin final to 0 as this should only be used for windows.
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/OuterWall.mo]: should be '1'

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo [hunk 3]
```diff
@@ -113,6 +113,27 @@ model RectangularZoneTemplate
     "Area fraction of the window frame of the ceiling"
     annotation(Dialog(tab="Ceiling", group="Window details",
     enable=hasWinCei));
+  parameter Real nWinA=1
+    "Scaling factor to model nWinA identical windows in facade A"
+    annotation(Dialog(tab="Face A", group="Window details",
+    enable=hasWinA));
+  parameter Real nWinB=0.15
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: should be '1'?

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo [hunk 4]
```diff
@@ -113,6 +113,27 @@ model RectangularZoneTemplate
...
+    "Scaling factor to model nWinB identical windows in facade B"
+    annotation(Dialog(tab="Face B", group="Window details",
+    enable=hasWinB));
+  parameter Real nWinC=0.15
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: should be '1'?

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo [hunk 5]
```diff
@@ -113,6 +113,27 @@ model RectangularZoneTemplate
...
+    "Scaling factor to model nWinC identical windows in facade C"
+    annotation(Dialog(tab="Face C", group="Window details",
+    enable=hasWinC));
+  parameter Real nWinD=0.15
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: should be '1'?

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo [hunk 6]
```diff
@@ -113,6 +113,27 @@ model RectangularZoneTemplate
...
+    "Scaling factor to model nWinD identical windows in facade D"
+    annotation(Dialog(tab="Face D", group="Window details",
+    enable=hasWinD));
+  parameter Real nWinCei=0.15
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: should be '1'?

File: IDEAS/Buildings/Components/SlabOnGround.mo [hunk 7]
```diff
@@ -141,6 +143,12 @@ zone that is surrounded by air at the ambient temperature.
 </html>", revisions="<html>
 <ul>
 <li>
+August 10, 2018 by Damien Picard:<br/>
+Set nWin final to 0 as this should only be used for windows.
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/SlabOnGround.mo]: should be '1'?

File: IDEAS/Buildings/Components/Window.mo [hunk 8]
```diff
@@ -13,8 +13,8 @@ model Window "Multipane window"
            dT_nominal=dT_nominal_a),
     QTra_design(fixed=false),
     Qgai(y=if sim.computeConservationOfEnergy then
-                                                  -(propsBus_a.surfCon.Q_flow +
-        propsBus_a.surfRad.Q_flow + propsBus_a.iSolDif.Q_flow + propsBus_a.iSolDir.Q_flow) else 0),
+                                                  -(propsBusInt.surfCon.Q_flow +
+        propsBusInt.surfRad.Q_flow + propsBusInt.iSolDif.Q_flow + propsBusInt.iSolDir.Q_flow) else 0),
```
Comment (Mathadon) [inline IDEAS/Buildings/Components/Window.mo]: the new implementation indeed seems correct to me

File: IDEAS/Buildings/Components/RectangularZoneTemplate.mo (hunk 3 above)
Comment (damienpicard) [reply IDEAS/Buildings/Components/RectangularZoneTemplate.mo]: !! Good you keep your eyes open!

File: IDEAS/Buildings/Components/Window.mo (hunk 8 above)
Comment (damienpicard) [reply IDEAS/Buildings/Components/Window.mo]: This lead to an error in conservation of energy as propsBusInt is an internal connector for which the sign of heat flow rates is not well defined. Using gain.propsBus_a instead solved the problem.

**Comment by Mathadon** [comment]
Still trouble in paradise. have to re-run PPD12 and 2 examples where results differ slightly.

**Comment by damienpicard** [comment]
zucht.

**Pull request merged**